tests/api_tests/nodes/test_views.py

```diff
@@ -863,6 +865,513 @@ def test_filtering_on_invalid_field(self):
+        assert_equal(res.json['data']['id'], self.user._id)
+
+    def test_get_private_node_contributor_detail_non_contributor(self):
+        res = self.app.get(self.private_url, auth=self.user_two.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_get_private_node_contributor_detail_not_logged_in(self):
+        res = self.app.get(self.private_url, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+    def test_get_private_node_non_contributor_detail_contributor_auth(self):
+        res = self.app.get(self.private_url_base.format(self.user_two._id), auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 404)
+
+    def test_get_private_node_invalid_user_detail_contributor_auth(self):
+        res = self.app.get(self.private_url_base.format('invalid'), auth=self.user.auth, expect_errors=True)
+        assert_equal(res.status_code, 404)
+
+
+class TestNodeContributorUpdate(ApiTestCase):
+    def setUp(self):
+        super(TestNodeContributorUpdate, self).setUp()
+        self.user = AuthUserFactory()
+        self.user_two = AuthUserFactory()
+
+        self.project = ProjectFactory(creator=self.user)
+        self.project.add_contributor(self.user_two, permissions=[permissions.READ, permissions.WRITE], visible=True, save=True)
+
+        self.url_creator = '/{}nodes/{}/contributors/{}/'.format(API_BASE, self.project._id, self.user._id)
+        self.url_contributor = '/{}nodes/{}/contributors/{}/'.format(API_BASE, self.project._id, self.user_two._id)
+
+    def test_change_contributor_permissions(self):
+        data = {
+            'permission': permissions.ADMIN,
+            'bibliographic': True
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        assert_equal(attributes['permission'], permissions.ADMIN)
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), [permissions.READ, permissions.WRITE, permissions.ADMIN])
+
+        data = {
+            'permission': permissions.WRITE,
+            'bibliographic': True
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        assert_equal(attributes['permission'], permissions.WRITE)
+
+        self.project.reload()
+        assert_equal(self.project.get_permissions(self.user_two), [permissions.READ, permissions.WRITE])
+
+        data = {
+            'permission': permissions.READ,
+            'bibliographic': True
+        }
+        res = self.app.put_json(self.url_contributor, data, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        attributes = res.json['data']['attributes']
+        assert_equal(attributes['permission'], permissions.READ)
```

Comment:
Can you add assertions to your tests that logs are created properly. This will help immensely with https://openscience.atlassian.net/browse/OSF-4445